src/Components/Web.JS/src/FileDownloader.ts

```diff
@@ -2,13 +2,42 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 export async function downloadFile(data: any, fileName: string) {
-    const arrayBuffer = await data.arrayBuffer();
-    const blob = new Blob([arrayBuffer]);
-    const url = URL.createObjectURL(blob);
-    const anchorElement = document.createElement('a');
-    anchorElement.href = url;
-    anchorElement.download = fileName;
-    anchorElement.click();
-    anchorElement.remove();
-    URL.revokeObjectURL(url);
+    // For Chromium browsers, show "Save As" dialog and then stream the data into the file without buffering it all in membey
+    if (typeof (window as any).showSaveFilePicker() === 'function')
+    {
+        // Show the "Save As" dialog
+        let fileWriter;
+        try {
+            const fileHandle = await (window as any).showSaveFilePicker();
+            // Create a FileSystemWritableFileStream to write to.
+            fileWriter = await fileHandle.createWritable();
+        } catch {
+            // User pressed cancel, so abort the whole thing
+            return;
+        }
+
+        const reader = data.getReader();
```

Comment:
Hmm is this working as-is? I was under the impression `data` is a `DotNetStream` here, hence we may have to do:

```suggestion
        const reader = await data.stream().getReader();
```

Reply:
This initially caused an unhandled exception. I believe `data` had to be converted into a `ReadableStream` in order for the `getReader()` method to work.

Reply:
Just to confirm, is `data` a `DotNetStream` here? If so, we can get the readable stream directly by calling `stream`:

https://github.com/dotnet/aspnetcore/blob/50aa2fd929750ff4e46df0150d96fbbb7788e5d0/src/JSInterop/Microsoft.JSInterop.JS/src/src/Microsoft.JSInterop.ts#L547-L552

Reply:
> This initially caused an unhandled exception.

What exception were you running into?

Reply:
When testing this with `FileDownloaderComponent.razor`, `data` is a `byte[]` instead of a `DotNetStreamReference`.

As a result, I was getting the following exception: `      TypeError: e.stream(...).getReader is not a function
`

Reply:
Hmm interesting, given:

https://github.com/dotnet/aspnetcore/pull/43076/files#diff-220a998a6be351c01be7cbb2fd3f549be1f0a2fe2f75444bdbd4be7394127fddR37

I thought it'd be a `DotNetStreamReference` here. I wonder what's going on.

Did you by change get to try out https://github.com/dotnet/aspnetcore/pull/43076/files/cd96575f2f252643bf0cdbe6361bddd1f3dd9819..7233f2fda2feebd040d65b60e9509e753fb80c15#diff-d538a65d016c0b3de7281bc088edb391759c410766dea79136741916f2233a8eR33? That currently is being treated as a `DotNetStreamReference`.

Reply:
The second option is working and `data` as a `DotNetStreamReference` is being saved to the file correctly.

The exception for `data.getReader()` is being handled, but no data is populating the file. I will look into this.